packages/apollo-server-core/src/requestPipeline.ts

```diff
@@ -284,11 +285,10 @@ export async function processGraphQLRequest<TContext>(
     let response: GraphQLResponse;
 
     try {
-      response = (await execute(
-        requestContext.document,
-        request.operationName,
-        request.variables,
-      )) as GraphQLResponse;
+      response = await execute(requestContext as WithRequired<
```

Comment:
TypeScript inquisition: Is it necessary to cast `requestContext` to having those required properties, or is that something that can be automatically inferred by the positional parameter typing on `execute` itself?

My believe here would have been that it was the latter and that we could remove this additional portion, but perhaps I mistaken.

Reply:
Unfortunately, the properties we set on `requestContext` won't change the type. So even if we know properties are there, TypeScript won't automatically narrow the type. I'm not sure what the restrictions on flow-dependent type narrowing are, because it does work in some other cases. But here we need to use an explicit type assertion.